### PR TITLE
Fix dashboard startup ordering.

### PR DESCRIPTION
**Pull request by mitchdenny**
We now do the reordering of the dashboard resource in `CreateExecutablesAsync(...)`. Reordering in the resources app model didn't do anything because the `_appResources` property ends up with a different ordering.

Note: this PR has a tempoary workaround for the AspireDashboardDir path not being correct. Need to figure out why that isn't working before merging. From what I can tell it could never work the way it is?

### CODE REVIEW

File: playground/eShopLite/AppHost/AppHost.csproj [hunk 1]
```diff
@@ -7,6 +7,7 @@
     <ImplicitUsings>enable</ImplicitUsings>
     <IsAspireHost>true</IsAspireHost>
     <AssemblyName>eShopLite.AppHost</AssemblyName>
+    <AspireDashboardDir>..\..\..\artifacts\bin\Aspire.Dashboard\Debug\net8.0\</AspireDashboardDir>
```
Comment (JamesNK) [inline playground/eShopLite/AppHost/AppHost.csproj:10]: Add TODO

**Review by JamesNK — APPROVED** [review]

**Review by drewnoakes — APPROVED** [review]

File: playground/eShopLite/AppHost/AppHost.csproj (hunk 1 above)
Comment (mitchdenny) [reply playground/eShopLite/AppHost/AppHost.csproj]: Moved this back up into the root `Directory.Build.props` (with a TODO).

File: Directory.Build.props [hunk 2]
```diff
@@ -36,7 +36,10 @@
     <BuildArch Condition=" '$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture)' == 'Arm64' ">arm64</BuildArch>
     <BuildArch Condition=" '$(BuildArch)' == '' ">amd64</BuildArch>
     <DcpDir>$(NuGetPackageRoot)microsoft.developercontrolplane.$(BuildOs)-$(BuildArch)/$(MicrosoftDeveloperControlPlanedarwinamd64PackageVersion)/tools/</DcpDir>
-    <AspireDashboardDir>$(DotNetOutputPath)Aspire.Dashboard/$(Configuration)/$(TargetFramework)/</AspireDashboardDir>
+
+    <!-- TODO: Need to figure out we can automatically detect target framework here. This property
+               is specified to support dashboard path metadata generation on the inner loop. -->
+    <AspireDashboardDir>$(MSBuildThisFileDirectory)/artifacts/bin/Aspire.Dashboard/$(Configuration)/net8.0/</AspireDashboardDir>
```
Comment (drewnoakes) [inline Directory.Build.props:42]: This `Directory.Build.props` file will be running before the `.csproj` file. Due to evaluation order, the `TargetFramework` property here won't be defined.

Options are to move this to a `.targets` file, or to a target.

Comment (mitchdenny) [reply Directory.Build.props:42]: Will tackle this in a follow up PR.

**Pull request merged**